runtime/src/accounts_db.rs

```diff
@@ -3316,24 +3318,20 @@ impl AccountsDb {
         max_root: Option<Slot>,
         clone_in_lock: bool,
     ) -> Option<(Slot, AppendVecId, usize, Option<LoadedAccountAccessor<'a>>)> {
-        let (lock, index) = match self.accounts_index.get(pubkey, Some(ancestors), max_root) {
-            AccountIndexGetResult::Found(lock, index) => (lock, index),
-            // we bail out pretty early for missing.
-            AccountIndexGetResult::NotFoundOnFork => {
-                return None;
-            }
-            AccountIndexGetResult::Missing(_) => {
-                return None;
-            }
-        };
-
-        let slot_list = lock.slot_list();
-        let (
-            slot,
-            AccountInfo {
-                store_id, offset, ..
-            },
-        ) = slot_list[index];
+        let (slot, store_id, offset) =
+            self.accounts_index
+                .get2(pubkey, Some(ancestors), max_root, |entry| {
+                    entry.map(|(lock, index)| {
+                        let slot_list = lock.slot_list();
+                        let (
+                            slot,
+                            AccountInfo {
+                                store_id, offset, ..
+                            },
+                        ) = slot_list[index];
+                        (slot, store_id, offset)
+                    })
+                })?;
```

Comment:
Noting for deeper review later

 The lock is released after this function call, before the call to `get_account_accessor()` below, whereas in the old code the lock was held throughout. 

The comment below "calling check_and_get_loaded_account is safe as long as we're guaranteed to hold the lock during the time and there should be no purge thanks to alive ancestors held by our caller" implies this may not be safe.